src/base64io/__init__.py

```diff
@@ -226,8 +226,8 @@ def _read_additional_data_removing_whitespace(self, data, total_bytes_to_read):
             # case the base64 module happily removes any whitespace.
             return data
 
-        _data_buffer = io.BytesIO()
-        _data_buffer.write(b"".join(data.split()))
+        _data_buffer = io.BytesIO() if isinstance(data, bytes) else io.StringIO()
```

Comment:
Method return type annotation needs to change to `AnyStr`.

Reply:
Done. The mypy tests are failing but I'm not sure why - I haven't used mypy previously. It looks like a false positive - maybe mypy is not able to handle the branching logic that handles bytes and str differently?

Reply:
The initial problem is just that you need to add an import for `AnyStr` in the `types` imports at the top. I'm not sure if the other issues it's running into are just a side effect of missing that or not.